yara-x/src/modules/macho/mod.rs

```diff
@@ -0,0 +1,2349 @@
+        }
+        _ => return Err(MachoError::UnsupportedCPUType),
+    }
+
+    // Swap bytes if neccessary
+    if should_swap_bytes(
+        macho_file
+            .magic
+            .ok_or(MachoError::MissingHeaderValue("magic".to_string()))?,
+    ) {
+        address = if is64 {
+            address.swap_bytes()
+        } else {
+            (address as u32).swap_bytes() as u64
+        };
+    }
+
+    // TODO: COMPILER FLAGS
+    macho_file.entry_point = macho_rva_to_offset(address, macho_file)?;
+
+    Ok(())
+}
+
+/// Handle the `LC_MAIN` command for Mach-O files.
+///
+/// This function processes the main command data, extracting the entry point for the Mach-O file
+/// and setting its stack size.
+///
+/// # Arguments
+///
+/// - `command_data`: The raw byte data of the main command.
+/// - `size`: The size of the main command data.
+/// - `macho_file`: Mutable reference to the protobuf representation of the Mach-O file.
+///
+/// # Returns
+///
+/// Returns a `Result<(), MachoError>` indicating the success or failure of the operation.
+fn handle_main(
+    command_data: &[u8],
+    size: usize,
+    macho_file: &mut File,
+) -> Result<(), MachoError> {
+    // Check size
+    if size < std::mem::size_of::<EntryPointCommand>() {
+        return Err(MachoError::FileSectionTooSmall(
+            "EntryPointCommand".to_string(),
+        ));
+    }
+
+    // Parse main command
+    let (_, mut entrypoint_cmd) = parse_entry_point_command(command_data)
+        .map_err(|e| MachoError::ParsingError(format!("{:?}", e)))?;
+
+    // Swap bytes if neccesarry
+    if should_swap_bytes(
+        macho_file
+            .magic
+            .ok_or(MachoError::MissingHeaderValue("magic".to_string()))?,
+    ) {
+        swap_entry_point_command(&mut entrypoint_cmd);
+    }
+
+    // TODO: COMPILER FLAGS
+    if false {
+        macho_file.entry_point =
+            macho_offset_to_rva(entrypoint_cmd.entryoff, macho_file)?;
+    } else {
+        macho_file.set_entry_point(entrypoint_cmd.entryoff);
+    }
+
+    macho_file.set_stack_size(entrypoint_cmd.stacksize);
+
+    Ok(())
+}
+
+/// Handle individual command segments based on the load command value.
+///
+/// This function processes various command segments in Mach-O files according to the specified load command value.
+/// It can handle segment commands for both 32-bit and 64-bit Mach-O files. Other commands like `LC_UNIXTHREAD` and `LC_MAIN`
+/// are also processed by this function.
+///
+/// # Arguments
+///
+/// - `cmd`: The load command type.
+/// - `cmdsize`: The size of the command data.
+/// - `command_data`: The raw byte data of the command.
+/// - `macho_file`: Mutable reference to the protobuf representation of the Mach-O file.
+/// - `process_segments`: Flag indicating if segment commands should be processed.
+///
+/// # Returns
+///
+/// Returns a `Result<u64, MachoError>` indicating the success or failure of the operation. On success, it returns the segment count.
```

Comment:
Keep lines under 80 characters long.

Reply:
Done, thank you. I have also added configuration to rustfmt that performs this automatically.